src/main/kotlin/com/terraformation/backend/accelerator/db/ApplicationStore.kt

```diff
@@ -95,6 +106,127 @@ class ApplicationStore(
+          .and(APPLICATION_MODULES.APPLICATION_ID.eq(applicationId))
+          .orderBy(MODULES.PHASE_ID, MODULES.ID)
+          .fetch { ApplicationModuleModel.of(it) }
+    }
+  }
+
+  fun fetchApplicationDeliverables(
+      organizationId: OrganizationId? = null,
+      projectId: ProjectId? = null,
+      applicationId: ApplicationId? = null,
+      deliverableId: DeliverableId? = null,
+      moduleId: ModuleId? = null,
+  ): List<DeliverableSubmissionModel> {
+    requirePermissions {
+      when {
+        projectId != null -> readProjectDeliverables(projectId)
+        applicationId != null -> readApplication(applicationId)
+        organizationId != null -> readOrganizationDeliverables(organizationId)
+        moduleId != null -> readModule(moduleId)
+        else -> readAllDeliverables()
+      }
+    }
+
+    val conditions =
+        listOfNotNull(
+            when {
+              projectId != null -> PROJECTS.ID.eq(projectId)
+              applicationId != null -> APPLICATIONS.ID.eq(applicationId)
+              organizationId != null -> ORGANIZATIONS.ID.eq(organizationId)
+              else -> null
+            },
+            deliverableId?.let { DELIVERABLES.ID.eq(it) },
+            moduleId?.let { DELIVERABLES.MODULE_ID.eq(it) },
+            DSL.or(
+                MODULES.PHASE_ID.eq(CohortPhase.PreScreen),
+                MODULES.PHASE_ID.eq(CohortPhase.Application),
+            ))
+
+    val documentsMultiset =
+        DSL.multiset(
+                DSL.select(SUBMISSION_DOCUMENTS.asterisk())
+                    .from(SUBMISSION_DOCUMENTS)
+                    .where(SUBMISSION_DOCUMENTS.SUBMISSION_ID.eq(SUBMISSIONS.ID))
+                    .orderBy(SUBMISSION_DOCUMENTS.ID))
+            .convertFrom { result -> result.map { SubmissionDocumentModel.of(it) } }
+
+    return dslContext
+        .select(
+            DELIVERABLE_DOCUMENTS.TEMPLATE_URL,
+            DELIVERABLES.DELIVERABLE_CATEGORY_ID,
+            DELIVERABLES.DELIVERABLE_TYPE_ID,
+            DELIVERABLES.DESCRIPTION_HTML,
+            DELIVERABLES.ID,
+            DELIVERABLES.MODULE_ID,
+            DELIVERABLES.NAME,
+            documentsMultiset,
+            MODULES.NAME,
+            ORGANIZATIONS.ID,
+            ORGANIZATIONS.NAME,
+            PROJECTS.ID,
+            PROJECTS.NAME,
+            SUBMISSIONS.FEEDBACK,
+            SUBMISSIONS.ID,
+            SUBMISSIONS.INTERNAL_COMMENT,
+            SUBMISSIONS.SUBMISSION_STATUS_ID,
+        )
+        .from(DELIVERABLES)
+        .join(MODULES)
+        .on(DELIVERABLES.MODULE_ID.eq(MODULES.ID))
+        .leftJoin(SUBMISSIONS)
+        .on(DELIVERABLES.ID.eq(SUBMISSIONS.DELIVERABLE_ID))
+        .leftJoin(DELIVERABLE_DOCUMENTS)
+        .on(DELIVERABLES.ID.eq(DELIVERABLE_DOCUMENTS.DELIVERABLE_ID))
+        .join(PROJECTS)
+        .on(SUBMISSIONS.PROJECT_ID.eq(PROJECTS.ID))
+        .leftJoin(APPLICATIONS)
+        .on(PROJECTS.ID.eq(APPLICATIONS.PROJECT_ID))
```

Comment:
Doesn't this need to join with `APPLICATION_MODULES` too? Otherwise it will return deliverables in modules that haven't been assigned to the application yet.